### PR TITLE
KAZOO-5545 + KAZOO-5548: save & access port requests timeline

### PR DESCRIPTION
**Pull request by fenollp**

### CODE REVIEW

File: applications/crossbar/src/modules/cb_port_requests.erl [hunk 1]
```diff
@@ -731,11 +736,39 @@ maybe_normalize_summary_results(Context, 'true') ->
...
+            Indexed = [{kz_json:get_integer_value(?METADATA_TIMESTAMP, JObj)
+                       ,kz_json:set_value(<<"type">>, timeline_event_type(JObj), JObj)
+                       }
+                       || JObj <- Comments ++ Transitions
+                      ],
+            {_, NewDoc} = lists:unzip(lists:keysort(1, Indexed)),
+            cb_context:set_resp_data(Context, NewDoc)
+    end.
+
+timeline_event_type(JObj) ->
```
Comment (k-anderson) [inline applications/crossbar/src/modules/cb_port_requests.erl]: @fenollp specs (and above)

File: applications/crossbar/doc/port_requests.md [hunk 2]
```diff
@@ -824,6 +826,81 @@ curl -v -X DELETE \
...
+
+Admins are able to list every comment and transitions regardless of their privacy setting.
+Non admins only see public comments and transitions.
+
+Private transitions can be listed by an authenticated user by setting the request value `show_private_transitions` to `true` (defaults to `false`).
+
+```shell
+curl -v -X GET \
+    -H "X-Auth-Token: {AUTH_TOKEN}" \
+    http://{SERVER}:8000/v2/accounts/{ACCOUNT_ID}/port_requests/{PORT_REQUEST_ID}/timeline?show_private_transitions=true
```
Comment (k-anderson) [inline applications/crossbar/doc/port_requests.md]: @fenollp what is a private transition?

File: applications/crossbar/doc/port_requests.md [hunk 3]
```diff
@@ -824,6 +826,81 @@ curl -v -X DELETE \
...
+```
+
+```json
+{
+    "auth_token": "{AUTH_TOKEN}",
+    "data": [
+        {
```
Comment (k-anderson) [inline applications/crossbar/doc/port_requests.md]: @fenollp Lets use nesting :)

{  
   "authorization":{  
      "reason":"manual",
      "account":{  
         "id":"{AUTH_ACCOUNT_ID}",
         "name":"{AUTH_ACCOUNT_NAME}"
      },
      "user":{  
         "first_name":"John",
         "last_name":"Doe",
         "id":"0d46906ff1eb36bff4d09b5b32fc14be"
      }
   },
   "type":"transition",
   "transition":{  
      "new":"unconfirmed",
      "previous":"undefined",
      "timestamp":63663993575
   }
}

File: applications/crossbar/src/modules_v2/cb_phone_numbers_v2.erl [hunk 4]
```diff
@@ -247,6 +247,7 @@ maybe_allow_updates(Context, _Nouns, _Verb) -> Context.
 -spec validate(cb_context:context(), path_token(), path_token()) -> cb_context:context().
 
 validate(Context) ->
+    lager:debug(">>> ~p", [cb_context:auth_user_id(Context)]),
```
Comment (k-anderson) [inline applications/crossbar/src/modules_v2/cb_phone_numbers_v2.erl]: @fenollp please remove :)

File: applications/crossbar/doc/port_requests.md [hunk 5]
```diff
@@ -832,7 +920,7 @@ curl -v -X DELETE \
 ```shell
 curl -v -X PATCH \
     -H "X-Auth-Token: {AUTH_TOKEN}" \
-    http://{SERVER}:8000/v2/accounts/{ACCOUNT_ID}/port_requests/{PORT_REQUEST_ID}/submitted
+    http://{SERVER}:8000/v2/accounts/{ACCOUNT_ID}/port_requests/{PORT_REQUEST_ID}/submitted?reason=this+was+approved+by+Jane+Doe
```
Comment (k-anderson) [inline applications/crossbar/doc/port_requests.md]: @fenollp why is this a query string param, should be in the payload (sorry if I missed this on the previous review)

Comment (fenollp) [inline applications/crossbar/doc/port_requests.md]: It can be in either: it's a req_value

**Pull request merged**